### PR TITLE
Add Travis Nightly

### PR DESCRIPTION
**Pull request by alexandrebouthinon**
## What does this PR do?
Add Travis nightly and optimize conditions

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -76,7 +100,7 @@ jobs:
           condition: $TRAVIS_BRANCH = master
 
     - stage: "Deploy Beta release on NPM"
-      if: type != pull_request && branch =~ ^[0-9]+-beta$
+      if:
```
Comment (Aschen) [inline .travis.yml]: Why keep this empty line?

Comment (alexandrebouthinon) [reply .travis.yml]: Oops

File: .travis.yml [hunk 2]
```diff
@@ -2,9 +2,9 @@ sudo: true
 
 jobs:
   include:
-    - name: Sdk tests
-      stage: Tests
-      sudo: false
+    - stage: Tests
+      name: "Unit Tests"
```
Comment (benoitvidis) [inline .travis.yml]:
```suggestion
      name: Unit Tests
```

File: .travis.yml [hunk 3]
```diff
@@ -22,21 +22,46 @@ jobs:
...
+        - npm run lint
+        - npm run unit-testing
         
       after_success:  
         - cat ./coverage/lcov.info | ./node_modules/.bin/codecov
 
-    - name: Documentation tests
-      stage: Tests
+    - stage: Tests
+      name: "Integration Tests"
```
Comment (benoitvidis) [inline .travis.yml]:
```suggestion
      name: Integration Tests
```

File: .travis.yml [hunk 4]
```diff
@@ -22,21 +22,46 @@ jobs:
...
+
+      install:
+        - npm install
+
+      script:
+        - npm run build
+        - npm run functional-testing
+        
+    - stage: Tests
+      name: "Documentation Tests"
```
Comment (benoitvidis) [inline .travis.yml]:
```suggestion
      name: Documentation Tests
```

File: .travis.yml [hunk 5]
```diff
@@ -2,9 +2,9 @@ sudo: true
 
 jobs:
   include:
-    - name: Sdk tests
-      stage: Tests
-      sudo: false
+    - stage: Tests
+      name: "Unit Tests"
+      if: type = pull_request OR (type = push && branch =~ /^master|[0-9]+-(dev|stable)$/) OR type = cron
```
Comment (benoitvidis) [inline .travis.yml]: nitpincking: unneeded parenthesis and `||`/`&&` mix with `AND`/`OR`
```suggestion
      if: type = pull_request OR type = push AND branch =~ /^master|[0-9]+-(dev|stable)$/ OR type = cron
```

File: .travis.yml [hunk 6]
```diff
@@ -22,21 +22,46 @@ jobs:
...
+        - npm run unit-testing
         
       after_success:  
         - cat ./coverage/lcov.info | ./node_modules/.bin/codecov
 
-    - name: Documentation tests
-      stage: Tests
+    - stage: Tests
+      name: "Integration Tests"
+      if: type = pull_request OR (type = push && branch =~ /^master|[0-9]+-(dev|stable)$/) OR type = cron
```
Comment (benoitvidis) [inline .travis.yml]:
```suggestion
      if: type = pull_request OR type = push AND branch =~ /^master|[0-9]+-(dev|stable)$/ OR type = cron
```

File: .travis.yml [hunk 7]
```diff
@@ -22,21 +22,46 @@ jobs:
...
+      install:
+        - npm install
+
+      script:
+        - npm run build
+        - npm run functional-testing
+        
+    - stage: Tests
+      name: "Documentation Tests"
+      if: type = pull_request OR (type = push && branch =~ /^master|[0-9]+-(dev|stable)$/) OR type = cron
```
Comment (benoitvidis) [inline .travis.yml]:
```suggestion
      if: type = pull_request OR type = push AND branch =~ /^master|[0-9]+-(dev|stable)$/ OR type = cron
```

File: .travis.yml [hunk 8]
```diff
@@ -46,7 +71,7 @@ jobs:
         - npm run doc-testing
 
     - stage: "Deploy Stable release on NPM"
```
Comment (benoitvidis) [inline .travis.yml]:
```suggestion
    - stage: Deploy Stable release on NPM
```

File: .travis.yml [hunk 9]
```diff
@@ -46,7 +71,7 @@ jobs:
...
-      if: type != pull_request && branch = master
+      if: tag IS present AND type != cron
```
Comment (benoitvidis) [inline .travis.yml]:
```suggestion
      if: branch = master AND tag IS present AND type != cron
```

File: .travis.yml [hunk 10]
```diff
@@ -80,7 +105,7 @@ jobs:
           condition: $TRAVIS_BRANCH = master
 
     - stage: "Deploy Beta release on NPM"
```
Comment (benoitvidis) [inline .travis.yml]:
```suggestion
    - stage: Deploy Beta release on NPM
```

**Pull request merged**